### PR TITLE
Refactor voice controllers

### LINKED ISSUE #280: Refactor voice controller
Factor out AV and Polly from RouteVoiceController. 

OOP - RouteVoiceController, AVVoiceController, PollyVoiceController
or
POP - Speakable protocol.

This will make it easier to make AWSPolly optional when that proposal is implemented.

@1ec5 @bsudekum

### PR DESCRIPTION
**Pull request by frederoni**
Fixes #280 

Split up voice controllers into RouteVoiceController and PollyVoiceController.

Replace AVSpeech with Polly:
```swift
navigationViewController.voiceController = PollyVoiceController(identityPoolId: "…")
```

@1ec5 @bsudekum 👀

### CODE REVIEW

File: MapboxNavigation/RouteVoiceController.swift [hunk 1]
```diff
@@ -162,19 +133,23 @@ public class RouteVoiceController: NSObject, AVSpeechSynthesizerDelegate {
     }
     
     func alertLevelDidChange(notification: NSNotification) {
-        guard isEnabled, volume > 0 else { return }
+        process(notification)
+    }
+    
+    func shouldSpeakFor(_ notification: NSNotification) -> Bool {
```
Comment (1ec5) [inline MapboxNavigation/RouteVoiceController.swift]: `shouldSpeak(for:)`

File: MapboxNavigation/RouteVoiceController.swift [hunk 2]
```diff
@@ -184,15 +159,16 @@ public class RouteVoiceController: NSObject, AVSpeechSynthesizerDelegate {
...
         
-        if useDefaultVoice {
-            speakFallBack(fallbackText)
-        } else {
-            speakWithPolly(speechString(notification: notification, markUpWithSSML: true))
-        }
+        return true
+    }
+    
+    func process(_ notification: NSNotification) {
```
Comment (1ec5) [inline MapboxNavigation/RouteVoiceController.swift]: This is a very vague name. Could we fold this into `alertLevelDidChange(_:)` and allow subclasses to override that method?

**Comment by frederoni** [comment]
Interesting idea about the category but less obvious to grasp and debug.

We should add a new target for PollyVoiceControllerwhen we create the podspec to 
provide modularity for both Carthage and CocoaPods users.

**Comment by 1ec5** [comment]
If we add a separate target, we'd have to add a separate podspec. Subspecs can't coexist with separate modules.

**Pull request merged**